localstack-core/localstack/aws/handlers/service.py

```diff
@@ -175,9 +177,20 @@ def create_exception_response(self, exception: Exception, context: RequestContex
         if operation and isinstance(exception, NotImplementedError):
             action_name = operation.name
             exception_message: str | None = exception.args[0] if exception.args else None
-            message = exception_message or get_coverage_link_for_service(service_name, action_name)
+            if exception_message is not None:
+                message = exception_message or get_coverage_link_for_service(
+                    service_name, action_name
+                )
+                error = CommonServiceException("InternalFailure", message, status_code=501)
+            else:
+                support_in_latest = get_aws_catalog().get_aws_service_status(
+                    service_name, action_name
+                )
+                error = map_catalog_availability_to_exception(
+                    service_name, action_name, support_in_latest
+                )
+                message = error.message
```

Comment:
question: I'm a bit confused by this. We only add the catalog info if there is no exception message already? Is this there to leave it open to service implementers to give custom messages in `NotImplementedError`s?